lib/mongo/collection/view.rb

```diff
@@ -132,7 +132,7 @@ def hash
       def initialize(collection, filter = {}, options = {})
         validate_doc!(filter)
         @collection = collection
-        parse_parameters!(BSON::Document.new(filter.freeze), BSON::Document.new(options.freeze))
+        parse_parameters!(BSON::Document.new(filter.dup.freeze), BSON::Document.new(options.dup.freeze))
```

Comment:
Thanks for pointing this out and for the proposed fix. I would prefer we freeze the @filter and @options variables here instead: https://github.com/matt-wormley/mongo-ruby-driver/blob/89d6016065c8b00ce0979da966872a0411245b64/lib/mongo/collection/view.rb#L162-L163

that way, we won't have to dup anything. What do you think?